libs/community/langchain_community/chat_models/premai.py

```diff
@@ -268,21 +251,34 @@ def _llm_type(self) -> str:
 
     @property
     def _default_params(self) -> Dict[str, Any]:
-        # FIXME: n and stop is not supported, so hardcoding to current default value
         return {
             "model": self.model,
             "system_prompt": self.system_prompt,
-            "top_p": self.top_p,
             "temperature": self.temperature,
-            "logit_bias": self.logit_bias,
             "max_tokens": self.max_tokens,
-            "presence_penalty": self.presence_penalty,
-            "frequency_penalty": self.frequency_penalty,
-            "seed": self.seed,
-            "stop": None,
+            "repositories": self.repositories,
         }
 
     def _get_all_kwargs(self, **kwargs: Any) -> Dict[str, Any]:
+        kwargs_to_ignore = [
+            "top_p",
+            "tools",
+            "frequency_penalty",
+            "presence_penalty",
+            "logit_bias",
+            "stop",
+            "seed",
+        ]
+        keys_to_remove = []
+
+        for key in kwargs:
+            if key in kwargs_to_ignore:
+                logger.warning(f"WARNING: Parameter {key} is not supported in kwargs.")
```

Comment:
can we make this warnings.warning? (logger.warning is for things user can't change, warnings.warning is for things user can change)

Reply:
oh I see, sure, let me do that

Reply:
Done